### PR TITLE
Feature/marketplace settlement request

### PR DESCRIPTION
**Pull request by prashantesmagico**

### CODE REVIEW

File: prisma/schema.prisma [hunk 1]
```diff
@@ -36,6 +37,21 @@ model Request {
   createdAt       DateTime          @default(now())
   updatedAt       DateTime          @updatedAt
   user            User              @relation(fields: [userId], references: [id]) // Specify fields and references here
+  Settlement      Settlement[]
+}
+
+model Settlement {
+  settlementId             Int                          @id @default(autoincrement())
+  requestId                Int                          @unique
+  userId                   Int
```
Comment (Ajay-Maury) [inline prisma/schema.prisma:62]: add admin ID as mentioned in docs

File: src/settlement/dto/create-settlement.dto.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  // Request ID associated with the request
+  @IsNotEmpty()
+  @IsInt()
+  requestId: number;
+
+  // User ID associated with the request.
+  @IsNotEmpty()
+  @IsInt()
+  userId: number;
+
```
Comment (Ajay-Maury) [inline src/settlement/dto/create-settlement.dto.ts:23]: adminId is missing

File: src/settlement/dto/response-settlement.dto.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import {
+  SettlementStatusEnum,
+  thirdPartyResponseStatusEnum,
+} from '../enum/settlement.enum';
+
+export class ResponseSettlementDto {
+  readonly SettlementId: number;
+  readonly requestId: number;
+  readonly userId: number;
```
Comment (Ajay-Maury) [inline src/settlement/dto/response-settlement.dto.ts:9]: add adminId

File: src/settlement/settlement.service.ts [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+      orderBy: {
+        [orderBy || 'createdAt']: 'asc', // Default sorting by createdAt
+      },
+      skip: offset,
+      take: limit,
+    });
+  }
+
+  // Get settlement for a specific user using the userId
+  async getAllSettlementForUser(userId: number, filter: any) {
```
Comment (Ajay-Maury) [inline src/settlement/settlement.service.ts]: define dto for filter instead of any

**Review by Ajay-Maury — CHANGES_REQUESTED** [review]

File: prisma/schema.prisma (hunk 1 above)
Comment (prashantesmagico) [reply prisma/schema.prisma:62]: added the Dummy admin model and relation

File: src/settlement/dto/create-settlement.dto.ts (hunk 2 above)
Comment (prashantesmagico) [reply src/settlement/dto/create-settlement.dto.ts:23]: added the changes

File: src/settlement/dto/response-settlement.dto.ts (hunk 3 above)
Comment (prashantesmagico) [reply src/settlement/dto/response-settlement.dto.ts:9]: added the changes

File: src/settlement/settlement.service.ts (hunk 4 above)
Comment (prashantesmagico) [reply src/settlement/settlement.service.ts]: done

File: prisma/schema.prisma [hunk 5]
```diff
@@ -36,6 +51,25 @@ model Request {
   createdAt       DateTime          @default(now())
   updatedAt       DateTime          @updatedAt
   user            User              @relation(fields: [userId], references: [id]) // Specify fields and references here
+  Admin           Admin?            @relation(fields: [adminId], references: [id])
+  adminId         Int?
+  Settlement      Settlement[]
```
Comment (SanchitEsMagico) [inline prisma/schema.prisma]: Can one request have multiple settlements?

**Review by SanchitEsMagico — APPROVED** [review]

File: prisma/schema.prisma (hunk 5 above)
Comment (prashantesmagico) [reply prisma/schema.prisma]: Already Implemented the error handler if the settlement exist for a request Id, new settlement for requestId throw an error

Comment (SanchitEsMagico) [reply prisma/schema.prisma]: Okay, but settlement shouldn't have been an array relationship

**Pull request merged**